diff -ur cyrus-imapd-2.5.10.orig/imap/tls.c cyrus-imapd-2.5.10/imap/tls.c
--- cyrus-imapd-2.5.10.orig/imap/tls.c	2016-10-17 21:01:04.000000000 +0000
+++ cyrus-imapd-2.5.10/imap/tls.c	2017-04-18 15:35:47.000000000 +0000
@@ -518,11 +518,13 @@
 	/* store the session in our database */
 
 	session_id = SSL_SESSION_get_id(sess, &session_id_length);
-	do {
-	    ret = cyrusdb_store(sessdb, (const char *) session_id,
-			    session_id_length,
-			    (const char *) data, len + sizeof(time_t), NULL);
-	} while (ret == CYRUSDB_AGAIN);
+	if (session_id_length) {
+	    do {
+	        ret = cyrusdb_store(sessdb, (const char *) session_id,
+			        session_id_length,
+			        (const char *) data, len + sizeof(time_t), NULL);
+	    } while (ret == CYRUSDB_AGAIN);
+	}
     }
 
     free(data);
@@ -551,6 +553,7 @@
     assert(id);
     assert(idlen <= SSL_MAX_SSL_SESSION_ID_LENGTH);
     
+    if (!idlen) return;    
     if (!sess_dbopen) return;
 
     do {
